kolibri/core/assets/src/views/sync/SelectDeviceModalGroup/SelectDeviceForm.vue

```diff
@@ -163,19 +164,22 @@
         deviceFilters.push(useDeviceChannelFilter({ id: props.filterByChannelId }));
       }
 
-      if (
-        props.filterByFacilityId !== null ||
-        props.filterByFacilityCanSignUp !== null ||
-        props.filterByOnMyOwnFacility !== null
-      ) {
+      const pickNotNull = v => v !== null;
+      // Either we build a facility filter or an empty object.
+      // Passing the empty object to useDeviceFacilityFilter is asking "are there ANY facilities?"
+      const facilityFilter = pickBy(
+        {
+          id: props.filterByFacilityId,
+          learner_can_sign_up: props.filterByFacilityCanSignUp,
+          on_my_own_setup: props.filterByOnMyOwnFacility,
+        },
+        pickNotNull,
+      );
+
+      // If we're filtering a particular facility
+      if (Object.keys(facilityFilter).length > 0 || props.filterByHasFacilities) {
         apiParams.subset_of_users_device = false;
```

Comment:
It maybe unexpected that we filter out SoUDs when applying `props.filterByHasFacilities`, but since it's a new prop, we don't have any use cases, so it's def fine until we do.